pypit/core/arskysub.py

```diff
@@ -80,33 +80,33 @@ def bg_subtraction_slit(slit, slitpix, edge_mask, sciframe, varframe, tilts,
     # All for evaluation
     all_slit = (slitpix == slit) & (~edge_mask)
 
-    # Pre-fit
-    pos_sky = (sky > 1.0) & (sky_ivar > 0.)
-    if np.sum(pos_sky) > ny:
-        lsky = np.log(sky[pos_sky])
-        lsky_ivar = lsky * 0. + 0.1
-
-        # Init bspline to get the sky breakpoints (kludgy)
-        tmp = pydl.bspline(wsky[pos_sky], nord=4, bkspace=bsp)
-
-        #skybkpt = bspline_bkpts(wsky[pos_sky], nord=4, bkspace=bsp $
-        #, / silent)
-        if False:
-            from matplotlib import pyplot as plt
-            plt.clf()
-            ax = plt.gca()
-            ax.scatter(wsky[pos_sky], lsky)
-            #ax.scatter(wsky[~full_out], sky[~full_out], color='red')
-            #ax.plot(wsky, yfit, color='green')
-            plt.show()
-            debugger.set_trace()
-        lskyset, outmask, lsky_fit, red_chi = arutils.bspline_profile(
-            wsky[pos_sky], lsky, lsky_ivar, np.ones_like(lsky),
-            fullbkpt = tmp.breakpoints, upper=sigrej, lower=sigrej,
-            kwargs_reject={'groupbadpix':True})
-        res = (sky[pos_sky] - np.exp(lsky_fit)) * np.sqrt(sky_ivar[pos_sky])
-        lmask = (res < 5.0) & (res > -4.0)
-        sky_ivar[pos_sky] = sky_ivar[pos_sky] * lmask
+    # Restrict fit to positive pixels only and mask out large outliers via a pre-fit to the log
+    if (POS_MASK==True):
```

Comment:
is True

I sure feel like I made these very same comments in a recent PR (now closed)